### PR TITLE
libelement: fix element_object_call API & uses

### PR DESCRIPTION
**Pull request by Zephilinox**

### CODE REVIEW

File: libelement.CLI/include/evaluate_command.hpp [hunk 1]
```diff
@@ -74,30 +74,33 @@ namespace libelement::cli
...
                     return compiler_message(error_conversion(result),
                                             "Failed to convert call expression to objects: " + expression + " called with " + custom_arguments.arguments + " at compile-time with element_result " + std::to_string(result),
                                             compilation_input.get_log_json());
                 }
 
                 result = element_object_call(expression_object, compilation_context, call_objects, call_objects_count, &result_object);
+
+                for (int i = 0; i < call_objects_count; ++i)
+                    element_delete_object(&call_objects[i]);
+
```
Comment (TheMunro) [inline libelement.CLI/include/evaluate_command.hpp:103]: Where are calling delete[] on call_objects?

**Review by TheMunro — CHANGES_REQUESTED** [review]

File: libelement.CLI/include/evaluate_command.hpp (hunk 1 above)
Comment (TheMunro) [reply libelement.CLI/include/evaluate_command.hpp:103]: Also, why are you newing stuff up before testing result != ELEMENT_OK? Could just new up before element_object_call

Comment (Zephilinox) [reply libelement.CLI/include/evaluate_command.hpp:103]: pushed a commit to fix the memory leak 😅

Comment (Zephilinox) [reply libelement.CLI/include/evaluate_command.hpp:100]: also changed it so we new after :)

**Review by TheMunro — APPROVED** [review]

**Pull request merged**